crates/scheduler/src/allocator.rs

```diff
@@ -262,3 +350,18 @@ where
         }
     }
 }
+
+fn current_max_price_index(selected: &[(PeerId, WorkerOfferRequest)]) -> Option<(usize, f64)> {
+    if selected.is_empty() {
+        return None;
+    }
+    let mut max_idx = 0;
+    let mut max_price = selected[0].1.price;
```

Comment:
```suggestion
    let mut max_price = selected[max_idx].1.price;
```
NIT: This could improve readability